### PR TITLE
TFE OIDC with GCP WIF blueprint added

### PR DESCRIPTION
**Pull request by averbuks**

### CODE REVIEW

**Review by juliocc — APPROVED** [review]
Really cool blueprint. Left a couple of comments below

File: blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+resource "google_iam_workload_identity_pool_provider" "tfe-pool-provider" {
+  project                            = module.project.project_id
+  workload_identity_pool_id          = google_iam_workload_identity_pool.tfe-pool.workload_identity_pool_id
+  workload_identity_pool_provider_id = var.workload_identity_pool_provider_id
+  display_name                       = "TFE Pool Provider"
+  description                        = "OIDC identity pool provider for TFE Integration"
+  # Use condition to make sure only token generated for a specific TFE Org and workspace can be used
+  attribute_condition = "attribute.terraform_workspace_id == \"${var.tfe_workspace_id}\" && attribute.terraform_organization_id == \"${var.tfe_organization_id}\""
+  attribute_mapping = {
```
[the remark this answers is not in the document]
Comment (averbuks) [reply blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf:54]: We can do so, but I don't see a reason to do mapping for the attributed we don't use in the conditions, in the example we only have org and workspace IDs in the conditions to make sure only a specific workflow can make use of it. 
Tho for a medium article I wanted to provide more visibility on other attributes.

[the remark this answers is not in the document]
Comment (juliocc) [reply blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf:54]: I think it's good idea to always map all the attributes from the provide, even you're currently not using them. Besides, we can have them documented here as our canonical reference ;)

[the remark this answers is not in the document]
Comment (averbuks) [reply blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf:54]: Done

File: blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+resource "google_iam_workload_identity_pool_provider" "tfe-pool-provider" {
+  project                            = module.project.project_id
+  workload_identity_pool_id          = google_iam_workload_identity_pool.tfe-pool.workload_identity_pool_id
+  workload_identity_pool_provider_id = var.workload_identity_pool_provider_id
+  display_name                       = "TFE Pool Provider"
+  description                        = "OIDC identity pool provider for TFE Integration"
+  # Use condition to make sure only token generated for a specific TFE Org and workspace can be used
+  attribute_condition = "attribute.terraform_workspace_id == \"${var.tfe_workspace_id}\" && attribute.terraform_organization_id == \"${var.tfe_organization_id}\""
```
[the remark this answers is not in the document]
Comment (averbuks) [reply blueprints/cloud-operations/terraform-enterprise-wif/gcp-workload-identity-provider/main.tf]: Done

**Pull request merged**